blueprints/component-addon/index.js

```diff
@@ -0,0 +1,59 @@
+var Blueprint   = require('../../lib/models/blueprint');
+var SilentError = require('../../lib/errors/silent');
+var stringUtil  = require('../../lib/utilities/string');
+var path        = require('path');
+
+module.exports = {
+  description: 'Generates a component. Name must contain a hyphen.',
+
+  fileMapTokens: function() {
+    return {
+      __path__: function(options) {
+        if (options.pod) {
+          return path.join(options.podPath, 'components', options.dasherizedModuleName);
+        }
+        return 'components';
+      },
+      __templatepath__: function(options) {
+        if (options.pod) {
+          return path.join(options.podPath, 'components', options.dasherizedModuleName);
+        }
+        return 'templates/components';
+      },
+      __templatename__: function(options) {
+        if (options.pod) {
+          return 'template';
+        }
+        return options.dasherizedModuleName;
+      }
+    };
+  },
+
+  normalizeEntityName: function(entityName) {
+    entityName = Blueprint.prototype.normalizeEntityName.apply(this, arguments);
+
+    if(! /\-/.test(entityName)) {
+      throw new SilentError('You specified "' + entityName + '", but in order to prevent ' +
+                            'clashes with current or future HTML element names, you must include ' +
+                            'a hyphen in the component name.');
+    }
+
+    return entityName;
+  },
+
+  locals: function(options) {
+    var addonEntity    = options.entity;
+    var addonRawName   = addonEntity.name;
+    var addonName      = stringUtil.dasherize(addonRawName);
+    var fileName       = stringUtil.dasherize(options.entity.name);
+    var pathName       = 'components';
+
+    if (options.pod) {
+      fileName = 'component';
+      pathName = path.join(addonName, 'components', fileName);
+    }
+    return {
+      modulePath: path.join(pathName, fileName)
```

Comment:
It may be worth adding this to the default locals in the Blueprint model. We'll be using this value for every -addon blueprint.